### PR TITLE
ARROW-16450: [Go][Docs] Include error handling in csv examples

### PR DESCRIPTION
**Pull request by wolfeidau**
As per the tests I have added checks in the examples as this tripped me up while dealing with nulls in my CSVs.

### CODE REVIEW

File: go/arrow/csv/reader_test.go [hunk 1]
```diff
@@ -61,6 +62,11 @@ func Example() {
 		n++
 	}
 
+	err := r.Err()
+	if err != nil {
+		log.Fatal(err)
+	}
```
Comment (zeroshade) [inline go/arrow/csv/reader_test.go:68]: I'd say add a comment here to be more informative and explicit about the error handling

Comment (wolfeidau) [reply go/arrow/csv/reader_test.go:70]: I have added a comment describing why we are checking the error.

File: go/arrow/csv/writer_test.go [hunk 2]
```diff
@@ -81,6 +81,11 @@ func Example_writer() {
 		n++
 	}
 
+	err = r.Err()
+	if err != nil {
+		log.Fatal(err)
+	}
```
[the remark this answers is not in the document]
Comment (wolfeidau) [reply go/arrow/csv/writer_test.go:89]: As above.

**Comment by wolfeidau** [comment]
I amended the commit rather than adding a new one, not sure if that is OK on this project.

**Pull request closed without merging**